mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -122,6 +123,92 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addCurrentKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getCurrentKillstreak();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setCurrentKillstreak(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+
+        this.updateCurrentKillstreak(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Sets the killstreak amouont. Infallible
+     * @param transaction represents the current transaction
+     * @param amount represents the amount of killstreak to set to
+     */
+    public void setCurrentKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        statisticsRecord.setCurrentKillstreak(amount);
+        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+
+        this.updateCurrentKillstreak(amount);
+    }
+
+    /**
+     * Adds highest killstreaks to the user account. Infallible.
+     * @param transaction represents the current transaction
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addHighestKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getHighestKillstreak();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setHighestKillstreak(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.HIGHEST_KILLSTREAK);
+
+        this.updateHighestKillstreak(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
```

Comment:
Think about what you need a highest killstreak value for. The highest killstreak needs to be updated if the latest killstreak is greater than the previous killstreak. I'd imagine you'd want a method like this:

```java
/**
 * If the given killstreak is higher than the current highest killstreak, updates the
 * highest killstreak to the given killstreak. Otherwise updates nothing.
 * 
 * @param killStreak a killstreak the player has recently achieved
 * @return the previous killstreak before any updates are carried out
 */
public int updateHighestKillstreak(Transaction transaction, int killStreak);
```

Reply:
answer above.